pkg/servicemonitor/servicemonitor.go

```diff
@@ -33,6 +34,22 @@ const (
 	UrlLabelName         string = "probe_url"
 )
 
+func (u *ServiceMonitor) TemplateAndUpdateServiceMonitorDeployment(routeURL, blackBoxExporterNamespace string, namespacedName types.NamespacedName, clusterID string) error {
+	params := map[string][]string{
+		// Currently we only support `http_2xx` as module
+		"module": {"http_2xx"},
+		"target": {routeURL},
+	}
+
+	if u.Hypershift == true {
```

Comment:
You can remove the else here

```suggestion

	if u.Hypershift == true {
		s := u.HyperShiftTemplateForServiceMonitorResource(routeURL, blackBoxExporterNamespace, params, namespacedName, clusterID)
		return u.HypershiftUpdateServiceMonitorDeployment(s)
	} 
	s := u.TemplateForServiceMonitorResource(routeURL, blackBoxExporterNamespace, params, namespacedName, clusterID)
	return u.UpdateServiceMonitorDeployment(s)
	
}
```

Reply:
Also we should only have one updateservicemoinitor deployment function. Similar to how you made DeleteServiceMonitorDeployment work for both Hypershift and non hypershift

Reply:
It's not possible to have one `TemplateServiceMonitor` nor `UpdateServiceMonitor` function because it interacts with a rhobsv1/monitoringv1 resource directly via its input/outputs.

`DeleteServiceMonitor` did not, which made it possible. Chaining `TemplateServiceMonitor` and `UpdateServiceMonitor` together enables this one function.

Reply:
i.e. `TemplateServiceMonitor` has non-relevant inputs, but outputs a ServiceMonitor
`UpdateServiceMonitor` takes a ServiceMonitor as an input and has non-relevant outputs
-->
`TemplateAndUpdateServiceMonitor` has non-relevant inputs/outputs, so it can hold logic inside it to pick the right servicemonitor